azext_iot/tests/iothub/configurations/test_iot_config_int.py

```diff
@@ -800,3 +804,96 @@ def test_device_configurations(self):
+                self.check("publisher", terms_publisherId),
+            ]
+
+            # Show IoT Edge module terms offer
+            self.cmd(
+                "iot edge image terms show --offer {} --plan {} --publisher {}".format(
+                    terms_offerId, terms_planId, terms_publisherId
+                ),
+                checks=offer_checks.append(self.check("accepted", "false"))
+            )
+
+            # Accept IoT Edge module terms offer
+            self.cmd(
+                "iot edge image terms accept --offer {} --plan {} --publisher {}".format(
+                    terms_offerId, terms_planId, terms_publisherId
+                ),
+                checks=offer_checks.append(self.check("accepted", "true"))
+            )
+
+            # Show the accepted IoT Edge module terms offer using URN
+            self.cmd(
+                "iot edge image terms show --urn {}".format(
+                    terms_urn
+                ),
+                checks=offer_checks.append(self.check("accepted", "true"))
+            )
+
+            # Cancel IoT Edge module terms offer
+            self.cmd(
+                "iot edge image terms cancel --offer {} --plan {} --publisher {}".format(
+                    terms_offerId, terms_planId, terms_publisherId
+                ),
+                checks=offer_checks.append(self.check("accepted", "false"))
+            )
+
+            # Error - providing offer, plan and publisher when URN is already provided
+            self.cmd(
+                "iot edge image terms show --urn {} --offer {} --plan {} --publisher {}".format(
+                    terms_urn, terms_offerId, terms_planId, terms_publisherId
+                ),
+                expect_failure=True,
+            )
+
+            # Error - invalid URN format
+            self.cmd(
+                "iot edge image terms show --urn {}".format(
+                    "bad_URN"
+                ),
+                expect_failure=True,
+            )
+
+            # Error - invalid offer
+            self.cmd(
+                "iot edge image terms show --offer {} --plan {} --publisher {}".format(
+                    "invalid_offer", terms_planId, terms_publisherId
+                ),
+                expect_failure=True,
+            )
+
+            # Error - invalid publisher
+            self.cmd(
+                "iot edge image terms show --offer {} --plan {} --publisher {}".format(
+                    terms_offerId, terms_planId, "invalid_publisher"
+                ),
+                expect_failure=True,
+            )
+        else:
+            # Error - Command run by a service principal user
+            self.cmd(
+                "iot edge image terms show --offer {} --plan {} --publisher {}".format(
+                    terms_offerId, terms_planId, terms_publisherId
+                ),
+                expect_failure=True,
+            )
+
+    @pytest.fixture(scope='class', autouse=True)
```

Comment:
I don't see a reason the fixture needs to be autouse, by applying it explicitly its easier to understand what fixtures are at play for what tests.

Reply:
Makes sense